sc-kpm/sc-python/services/http_api/auth/admin_handlers.py

```diff
@@ -0,0 +1,92 @@
+import json
+
+from http_api.auth import constants as cnt
+from http_api.auth.base_handler import BaseHandler
+from http_api.auth.common import get_response_message
+from http_api.auth.database import DataBase
+from http_api.auth.validators import TokenValidator, TokenType
+from http_api.auth.verifiers import username_verifier
+
+
+def _verify_user_info_in_database(database: DataBase, name: str, password: str, role_id: str) -> str:
+    if not username_verifier.verify(name):
+        message_desc = cnt.MSG_INVALID_USERNAME
+    elif len(password.strip()) == 0:
+        message_desc = cnt.MSG_INVALID_PASSWORD
+    elif database.is_such_user_in_base(name):
+        message_desc = cnt.MSG_USER_IS_IN_BASE
+    elif not database.is_such_role_in_base(role_id):
+        message_desc = cnt.MSG_INVALID_ROLE
+    else:
+        message_desc = cnt.MSG_ALL_DONE
+    return message_desc
+
+
+class UserHandler(BaseHandler):
+    @TokenValidator.validate_typed_token(TokenType.ACCESS)
+    def post(self) -> None:
+        """ Add new user """
+        database = DataBase()
+        request_params = self._get_request_params([cnt.NAME, cnt.PASSWORD, cnt.ROLE_ID])
+        msg_desc = _verify_user_info_in_database(
+            database,
+            name=request_params[cnt.NAME],
+            password=request_params[cnt.PASSWORD],
+            role_id=request_params[cnt.ROLE_ID]
+        )
+        response = get_response_message(msg_desc)
+        if msg_desc == cnt.MSG_ALL_DONE:
+            database.add_user(**request_params)
+        self.write(response)
+
+    @TokenValidator.validate_typed_token(TokenType.ACCESS)
+    def get(self) -> None:
+        """ Get info about user """
+        database = DataBase()
+        request_params = self._get_request_params([cnt.ID])
+        human_info = database.get_user_by_id(**request_params)
+        if human_info is not None:
+            response = json.dumps({
+                cnt.ID: human_info[cnt.ID],
+                cnt.NAME: human_info[cnt.NAME],
+                cnt.ROLE: human_info[cnt.ROLE]
+            })
+        else:
+            response = get_response_message(cnt.MSG_USER_NOT_FOUND)
+        self.write(response)
+
+    @TokenValidator.validate_typed_token(TokenType.ACCESS)
+    def delete(self) -> None:
+        """ Delete user """
+        database = DataBase()
+        request_params = self._get_request_params([cnt.ID])
+        database.delete_user_by_id(**request_params)
+        response = get_response_message(cnt.MSG_ALL_DONE)
+        self.write(response)
+
+    @TokenValidator.validate_typed_token(TokenType.ACCESS)
+    def put(self) -> None:
+        """ Update user """
+        database = DataBase()
```

Comment:
What should admin change in this request? Right now admin cannot change single role of user, he should change also username. I think it's a bug
![image](https://user-images.githubusercontent.com/34061914/158572233-fcc0cb9a-2427-4e53-afe3-a2efa2d50843.png)
![image](https://user-images.githubusercontent.com/34061914/158572323-e7d1eeaf-e0d8-4bfa-b35f-1934ad64fcbc.png)

Reply:
Also PUT method return "msg_code": 0 even if user not exist in DB. If I cannot create users with this method, so better to return message code 4 like in GET method